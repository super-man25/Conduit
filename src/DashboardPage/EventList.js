import styled from 'styled-components';

const EventList = styled.div`
  height: calc(100% - 176px);
  width: auto;
  margin: 0;
  padding: 30px;
  ::before {
    content: 'Event List Here...';
  }
`;

export default EventList;