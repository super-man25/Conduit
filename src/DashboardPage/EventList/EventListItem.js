import React from 'react';
import styled from 'styled-components';
import { cssConstants } from '../../_constants';
import { isPast } from 'date-fns';
import { darken } from 'polished';

import {
  Flex,
  Spacing,
  H4,
  P2
} from '../../_components';

import {
  readableDuration,
  readableDate,
  orDash
} from '../../_helpers/stringUtils';

const Heading = H4.extend`
  margin: 0;
  padding: 0;
`;

const Container = styled.div`
  padding: 16px 32px 16px 40px;
  border-bottom: 1px solid ${cssConstants.PRIMARY_LIGHT_GRAY};
  background-color: ${(props) => { return props.past ? cssConstants.PRIMARY_LIGHTER_GRAY : cssConstants.PRIMARY_WHITE; }};
  color: ${(props) => { return props.past ? cssConstants.PRIMARY_DARK_GRAY : cssConstants.PRIMARY_BLACK; }};
  transition: 0.15s ease-in-out all;
  position: relative;

  :hover {
    cursor: pointer;
    background-color: ${(props) => darken(0.05, (props.past ? cssConstants.PRIMARY_LIGHTER_GRAY : cssConstants.PRIMARY_WHITE))};
  }

  ::before {
    transition: 0.1s ease-in-out width;
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    width: ${(props) => { return props.active ? '16px' : '0px'; }};
    background-color: ${cssConstants.PRIMARY_DARK_BLUE};
  }
`;

export const EventListItem = (props) => {
  const { event, active, onClick } = props;

  return (
    <Container onClick={onClick} active={active} past={isPast(event.timestamp)}>
      <Flex direction="row" justify="space-between">
        <P2 color={cssConstants.PRIMARY_GRAY}>GAME SCORE: {orDash(event.score)}</P2>
        <P2 color={cssConstants.PRIMARY_GRAY}>updated {readableDuration(event.modifiedAt)} ago</P2>
      </Flex>
      <Spacing height="20px" />
      <Heading>{event.name}</Heading>
      <Spacing height="8px" />
      <Flex direction="row" justify="space-between">
        <P2>{readableDate(event.timestamp)}</P2>
        <P2>QUANTITY: {orDash(event.inventory)}/{orDash(event.capacity)}</P2>
      </Flex>
    </Container>
  );
};

EventListItem.defaultProps = {
  onClick: () => {},
  event: {},
  active: false
};
