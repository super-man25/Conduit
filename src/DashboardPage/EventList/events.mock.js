import { addDays } from 'date-fns';

export const events = [
  {
    id: 1,
    createdAt: new Date(),
    modifiedAt: new Date('2018-04-12T20:45:23+00:00'),
    timestamp: addDays(new Date(), 1),
    promotion: true,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 2,
    createdAt: new Date(),
    modifiedAt: new Date(),
    timestamp: addDays(new Date(), 1),
    promotion: false,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 3,
    createdAt: new Date(),
    modifiedAt: new Date(),
    timestamp: addDays(new Date(), 1),
    promotion: false,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 4,
    createdAt: new Date(),
    modifiedAt: new Date(),
    timestamp: new Date(),
    promotion: false,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 5,
    createdAt: new Date(),
    modifiedAt: new Date(),
    timestamp: new Date(),
    promotion: true,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 6,
    createdAt: new Date(),
    modifiedAt: new Date('2018-04-12T20:45:23+00:00'),
    timestamp: new Date('2018-04-12T20:45:23+00:00'),
    promotion: true,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 7,
    createdAt: new Date(),
    modifiedAt: new Date(),
    timestamp: new Date(),
    promotion: false,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 8,
    createdAt: new Date(),
    modifiedAt: new Date(),
    timestamp: new Date(),
    promotion: false,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 9,
    createdAt: new Date(),
    modifiedAt: new Date(),
    timestamp: new Date(),
    promotion: false,
    name: 'St. Louis Cardinals @ New York Mets'
  }, {
    id: 10,
    createdAt: new Date(),
    modifiedAt: new Date(),
    timestamp: new Date(),
    promotion: true,
    name: 'St. Louis Cardinals @ New York Mets'
  }
];
