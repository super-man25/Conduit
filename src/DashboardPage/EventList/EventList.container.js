import React from 'react';
import { EventList as Presenter } from './EventList.presenter';
import { events } from './events.mock';

const filterOptions = [
  {
    id: 1,
    label: 'All Events'
  }, {
    id: 2,
    label: 'Promotions'
  }
];

export class EventList extends React.Component {
  state = {
    active: -1,
    selectedFilter: 0,
    timestampSort: 'asc'
  }

  render() {
    const { active, selectedFilter, timestampSort } = this.state;

    return (
      <Presenter
        onClick={(a) => this.setState({ active: a })}
        active={active}
        events={events}

        filterOptions={filterOptions}
        onFilterSelect={(o, i) => this.setState({ selectedFilter: i })}
        selectedFilter={selectedFilter}

        timestampSort={timestampSort}
        onTimestampSortChange={(curr, next) => this.setState({ timestampSort: next })}

        // loading
      />
    );
  }
}
