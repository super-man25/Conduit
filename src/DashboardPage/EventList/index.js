import React from 'react';
import { EventList } from './EventList';
import { events } from './events.mock';


const EL = () => (
  <EventList
    events={events}
    // loading
  />
);

export { EL as EventList };
