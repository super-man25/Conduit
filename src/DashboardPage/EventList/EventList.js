import React from 'react';
import styled from 'styled-components';
import { EventListItem } from './EventListItem';
import { cssConstants } from '../../_constants';

import {
  H4,
  Input,
  Loader,
  Spacing
} from '../../_components';

const Heading = H4.extend`
  margin: 0;
  padding: 0;
`;

const HeaderContainer = styled.div`
  border-bottom: 1px solid ${cssConstants.PRIMARY_LIGHT_GRAY};
`;

const NoContentWrap = styled.div`
  padding: 10vh 0;
  text-align: center;
  color: ${cssConstants.PRIMARY_DARK_GRAY};
`;

export class EventList extends React.Component {
  state = { active: -1 };

  render() {
    const { loading, events } = this.props;
    const { active } = this.state;

    return (
      <div>
        <HeaderContainer>
          <Spacing padding="24px 40px">
            <Heading>All Events</Heading>
            <Input type="text" />
          </Spacing>
        </HeaderContainer>
        <div>
          {!!loading && <Spacing padding="10vh 0">
            <Loader />
          </Spacing>}

          {!loading && !events.length && <NoContentWrap>
            <Heading>No events matching query</Heading>
          </NoContentWrap>}

          {!loading && !!events.length && events.map((event, key) => (
            <EventListItem
              onClick={() => this.setState({ active: key })}
              active={key === active}
              key={key}
              event={event}
            />
          ))}
        </div>
      </div>
    );
  }
}

EventList.defaultProps = {
  events: [],
  loading: false
};

