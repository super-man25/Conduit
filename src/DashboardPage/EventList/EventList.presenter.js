import React from 'react';
import styled from 'styled-components';
import { EventListItem } from './EventListItem';
import { cssConstants } from '../../_constants';
import { PropTypes } from 'prop-types';
import { Event as EventModel } from '../../_models';

import {
  Flex,
  H4,
  Input,
  Loader,
  SortableButton,
  DropdownButton,
  Spacing
} from '../../_components';

const Heading = H4.extend`
  margin: 0;
  padding: 0;
`;

const HeaderContainer = styled.div`
  border-bottom: 1px solid ${cssConstants.PRIMARY_LIGHT_GRAY};
`;

const NoContentWrap = styled.div`
  padding: 10vh 0;
  text-align: center;
  color: ${cssConstants.PRIMARY_DARK_GRAY};
`;

export const EventList = (props) => {
  const {
    active,
    events,
    loading,
    onClick,
    filterOptions,
    selectedFilter,
    onFilterSelect,
    timestampSort,
    onTimestampSortChange
  } = props;

  return (
    <div>
      <HeaderContainer>
        <Spacing padding="24px 40px">
          <Heading>All Events</Heading>
          <Input type="text" placeholder="Search" />
          <Spacing padding="4px 0 0">
            <Flex direction="row" justify="space-between">
              <SortableButton
                direction={timestampSort}
                onClick={onTimestampSortChange}
              >
                Upcoming
              </SortableButton>
              <DropdownButton
                options={filterOptions}
                parseLabel={(o) => o.label}
                selected={selectedFilter}
                onSelect={onFilterSelect}
              />
            </Flex>
          </Spacing>
        </Spacing>
      </HeaderContainer>
      <div>
        {!!loading && <Spacing padding="10vh 0">
          <Loader />
        </Spacing>}

        {!loading && !events.length && <NoContentWrap>
          <Heading>No events matching query</Heading>
        </NoContentWrap>}

        {!loading && !!events.length && events.map((event, key) => (
          <EventListItem
            onClick={() => onClick(key, event)}
            active={key === active}
            key={key}
            event={event}
          />
        ))}
      </div>
    </div>
  );
};

EventList.defaultProps = {
  events: [],
  loading: false,
  onClick: () => {}
};

EventList.propTypes = {
  /** Index of the currently selected event. Should be -1 if no event is active. */
  active: PropTypes.number,

  /** List of events */
  events: PropTypes.arrayOf(EventModel),

  /** Whether or not the list is loading. Defaults to false. */
  loading: PropTypes.bool,

  /** Callback fired when a list item is clicked. Called with the event item key and object. */
  onClick: PropTypes.func,

  /** Options for the filter */
  filterOptions: DropdownButton.propTypes.options,

  /** The selected filter */
  selectedFilter: DropdownButton.propTypes.selected,

  /** Callback fired when a filter is selected */
  onFilterSelect: DropdownButton.propTypes.onSelect,

  /** Sort direction for timestamp */
  timestampSort: SortableButton.propTypes.direction,

  /** Callback fired when the timestamp sort direction is changed */
  onTimestampSortChange: SortableButton.propTypes.onClick
};
