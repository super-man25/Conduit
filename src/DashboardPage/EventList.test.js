import React from 'react';
import EventList from './EventList';
import renderer from 'react-test-renderer';

xit('renders correctly with no props', () => {
  const tree = renderer
    .create(<EventList />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
