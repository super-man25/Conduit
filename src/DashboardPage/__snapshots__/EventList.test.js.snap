// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no props 1`] = `
<div>
  <div
    className="sc-cHGsZl kjQnlZ"
  >
    <div
      className="sc-fMiknA hlNxzZ"
    >
      <h4
        className="sc-kjoXOD gHaHcG"
      >
        2018 Season
      </h4>
      <div
        className="sc-fMiknA hFCTbF"
        height="4px"
      />
      <input
        className="sc-hMqMXs kIiiqr"
        placeholder="Search"
        type="text"
      />
      <div
        className="sc-fMiknA hFCTbF"
        height="4px"
      />
      <div
        className="sc-kgoBCf fOSLLy"
        direction="row"
      >
        <div
          className="sc-bRBYWo bhewGY"
        >
          <div
            className="sc-kgoBCf bOvlAE"
            direction="row"
          >
            <p
              className="sc-hzDkRC hTSQNn"
              onClick={[Function]}
            >
              Upcoming
            </p>
            <div
              className="sc-jhAzac hmzSPl"
            >
              <div
                className="sc-fBuWsC bIToNP"
              >
                <svg
                  className="sc-chPdSV hncgqc"
                  color="#808080"
                  viewBox="0 0 24 24"
                  width="24px"
                >
                  <path
                    d="M7 14l5-5 5 5z"
                  />
                </svg>
              </div>
              <div
                className="sc-fBuWsC ewaGuK"
              >
                <svg
                  className="sc-chPdSV gADjDG"
                  color="#000"
                  viewBox="0 0 24 24"
                  width="24px"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="sc-kpOJdX kkGiTF"
          >
            <div
              className="sc-kgoBCf bOvlAE"
              direction="row"
            >
              <p
                className="sc-dxgOiQ hwxOEL"
                onClick={[Function]}
              >
                All Events
              </p>
              <svg
                className="sc-chPdSV gADjDG"
                color="#000"
                viewBox="0 0 24 24"
                width="24px"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <div
              className="sc-ckVGcZ gwIFHO"
            >
              <div
                className="sc-jKJlTe kFHFms"
                onClick={[Function]}
              >
                All Events
              </div>
              <div
                className="sc-jKJlTe kFHFms"
                onClick={[Function]}
              >
                Promotions
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="sc-gisBJw fWRUul"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            12 days
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Wed, 4/25/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      >
        <div
          className="sc-kgoBCf eLedlk"
          direction="column"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            promotion
          </p>
          <div
            className="sc-fMiknA fQrxJP"
            height="8px"
          />
          <svg
            className="sc-chPdSV lYzpY"
            color={undefined}
            viewBox="0 0 24 24"
            width="28px"
          >
            <path
              d="M9.01496516,15.424523 L8.26444368,21.0751434 L11.4855042,19.1425071 C11.8021866,18.9524976 12.1978134,18.9524976 12.5144958,19.1425071 L15.7358817,21.0753387 L14.9861004,15.4240941 C14.0634431,15.7955523 13.0555972,16 12,16 C10.9448091,16 9.93733283,15.7957097 9.01496516,15.424523 Z M7.13949704,14.3546913 C5.23083455,12.8926088 4,10.5900428 4,8 C4,3.581722 7.581722,0 12,0 C16.418278,0 20,3.581722 20,8 C20,10.5895337 18.7696493,12.8917035 16.8616285,14.353829 L17.9913131,22.8684771 C18.1014481,23.6985855 17.2035539,24.2883227 16.4855042,23.8574929 L12,21.1661904 L7.51449576,23.8574929 C6.79638903,24.288357 5.89844315,23.6984938 6.0087057,22.8683352 L7.13949704,14.3546913 Z M15.3305913,12.9914684 C16.9401106,11.9153689 18,10.0814597 18,8 C18,4.6862915 15.3137085,2 12,2 C8.6862915,2 6,4.6862915 6,8 C6,11.3137085 8.6862915,14 12,14 C13.1638413,14 14.2502872,13.6686308 15.170137,13.0950932 C15.2198314,13.0557718 15.2734995,13.0209907 15.3305913,12.9914684 Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="sc-gisBJw fWRUul"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            less than a minute
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Wed, 4/25/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      />
    </div>
    <div
      className="sc-gisBJw fWRUul"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            less than a minute
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Wed, 4/25/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      />
    </div>
    <div
      className="sc-gisBJw hVXddK"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            less than a minute
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Tue, 4/24/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      />
    </div>
    <div
      className="sc-gisBJw hVXddK"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            less than a minute
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Tue, 4/24/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      >
        <div
          className="sc-kgoBCf eLedlk"
          direction="column"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            promotion
          </p>
          <div
            className="sc-fMiknA fQrxJP"
            height="8px"
          />
          <svg
            className="sc-chPdSV fwbhwV"
            color="#767676"
            viewBox="0 0 24 24"
            width="28px"
          >
            <path
              d="M9.01496516,15.424523 L8.26444368,21.0751434 L11.4855042,19.1425071 C11.8021866,18.9524976 12.1978134,18.9524976 12.5144958,19.1425071 L15.7358817,21.0753387 L14.9861004,15.4240941 C14.0634431,15.7955523 13.0555972,16 12,16 C10.9448091,16 9.93733283,15.7957097 9.01496516,15.424523 Z M7.13949704,14.3546913 C5.23083455,12.8926088 4,10.5900428 4,8 C4,3.581722 7.581722,0 12,0 C16.418278,0 20,3.581722 20,8 C20,10.5895337 18.7696493,12.8917035 16.8616285,14.353829 L17.9913131,22.8684771 C18.1014481,23.6985855 17.2035539,24.2883227 16.4855042,23.8574929 L12,21.1661904 L7.51449576,23.8574929 C6.79638903,24.288357 5.89844315,23.6984938 6.0087057,22.8683352 L7.13949704,14.3546913 Z M15.3305913,12.9914684 C16.9401106,11.9153689 18,10.0814597 18,8 C18,4.6862915 15.3137085,2 12,2 C8.6862915,2 6,4.6862915 6,8 C6,11.3137085 8.6862915,14 12,14 C13.1638413,14 14.2502872,13.6686308 15.170137,13.0950932 C15.2198314,13.0557718 15.2734995,13.0209907 15.3305913,12.9914684 Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="sc-gisBJw hVXddK"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            12 days
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Thu, 4/12/18 @ 3:45PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      >
        <div
          className="sc-kgoBCf eLedlk"
          direction="column"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            promotion
          </p>
          <div
            className="sc-fMiknA fQrxJP"
            height="8px"
          />
          <svg
            className="sc-chPdSV fwbhwV"
            color="#767676"
            viewBox="0 0 24 24"
            width="28px"
          >
            <path
              d="M9.01496516,15.424523 L8.26444368,21.0751434 L11.4855042,19.1425071 C11.8021866,18.9524976 12.1978134,18.9524976 12.5144958,19.1425071 L15.7358817,21.0753387 L14.9861004,15.4240941 C14.0634431,15.7955523 13.0555972,16 12,16 C10.9448091,16 9.93733283,15.7957097 9.01496516,15.424523 Z M7.13949704,14.3546913 C5.23083455,12.8926088 4,10.5900428 4,8 C4,3.581722 7.581722,0 12,0 C16.418278,0 20,3.581722 20,8 C20,10.5895337 18.7696493,12.8917035 16.8616285,14.353829 L17.9913131,22.8684771 C18.1014481,23.6985855 17.2035539,24.2883227 16.4855042,23.8574929 L12,21.1661904 L7.51449576,23.8574929 C6.79638903,24.288357 5.89844315,23.6984938 6.0087057,22.8683352 L7.13949704,14.3546913 Z M15.3305913,12.9914684 C16.9401106,11.9153689 18,10.0814597 18,8 C18,4.6862915 15.3137085,2 12,2 C8.6862915,2 6,4.6862915 6,8 C6,11.3137085 8.6862915,14 12,14 C13.1638413,14 14.2502872,13.6686308 15.170137,13.0950932 C15.2198314,13.0557718 15.2734995,13.0209907 15.3305913,12.9914684 Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="sc-gisBJw hVXddK"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            less than a minute
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Tue, 4/24/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      />
    </div>
    <div
      className="sc-gisBJw hVXddK"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            less than a minute
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Tue, 4/24/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      />
    </div>
    <div
      className="sc-gisBJw hVXddK"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            less than a minute
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Tue, 4/24/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      />
    </div>
    <div
      className="sc-gisBJw hVXddK"
      onClick={[Function]}
    >
      <div
        className="sc-kGXeez cWJdvh"
      >
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            GAME SCORE: 
            --
          </p>
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            updated 
            less than a minute
             ago
          </p>
        </div>
        <div
          className="sc-fMiknA kZgHio"
          height="16px"
        />
        <h4
          className="sc-caSCKo dLmmTk"
        >
          St. Louis Cardinals @ New York Mets
        </h4>
        <div
          className="sc-fMiknA fQrxJP"
          height="8px"
        />
        <div
          className="sc-kgoBCf fOSLLy"
          direction="row"
        >
          <p
            className="sc-htoDjs fXGNOh"
          >
            Tue, 4/24/18 @ 12:09PM
          </p>
          <p
            className="sc-htoDjs fXGNOh"
          >
            QUANTITY: 
            --
            /
            --
          </p>
        </div>
      </div>
      <div
        className="sc-fMiknA htWfub"
        width="16px"
      />
      <div
        className="sc-fMiknA dxRBnk"
        width="60px"
      >
        <div
          className="sc-kgoBCf eLedlk"
          direction="column"
        >
          <p
            className="sc-htoDjs hGdgwo"
            color="#999999"
          >
            promotion
          </p>
          <div
            className="sc-fMiknA fQrxJP"
            height="8px"
          />
          <svg
            className="sc-chPdSV fwbhwV"
            color="#767676"
            viewBox="0 0 24 24"
            width="28px"
          >
            <path
              d="M9.01496516,15.424523 L8.26444368,21.0751434 L11.4855042,19.1425071 C11.8021866,18.9524976 12.1978134,18.9524976 12.5144958,19.1425071 L15.7358817,21.0753387 L14.9861004,15.4240941 C14.0634431,15.7955523 13.0555972,16 12,16 C10.9448091,16 9.93733283,15.7957097 9.01496516,15.424523 Z M7.13949704,14.3546913 C5.23083455,12.8926088 4,10.5900428 4,8 C4,3.581722 7.581722,0 12,0 C16.418278,0 20,3.581722 20,8 C20,10.5895337 18.7696493,12.8917035 16.8616285,14.353829 L17.9913131,22.8684771 C18.1014481,23.6985855 17.2035539,24.2883227 16.4855042,23.8574929 L12,21.1661904 L7.51449576,23.8574929 C6.79638903,24.288357 5.89844315,23.6984938 6.0087057,22.8683352 L7.13949704,14.3546913 Z M15.3305913,12.9914684 C16.9401106,11.9153689 18,10.0814597 18,8 C18,4.6862915 15.3137085,2 12,2 C8.6862915,2 6,4.6862915 6,8 C6,11.3137085 8.6862915,14 12,14 C13.1638413,14 14.2502872,13.6686308 15.170137,13.0950932 C15.2198314,13.0557718 15.2734995,13.0209907 15.3305913,12.9914684 Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
